projects/storefrontlib/src/cms-structure/outlet/outlet.service.ts

```diff
@@ -28,43 +20,49 @@ export class OutletService {
    * @param template the `TemplateRef` that will be used to insert UI
    * @param position the `OutletPosition` in the UI
    */
-  add(
-    outlet: string,
-    template: TemplateRef<any>,
-    position?: OutletPosition
-  ): void;
+  add(outlet: string, template: T, position?: OutletPosition): void;
   /**
    * @param factory The `ComponentFactory` that will be dynamically added to the outlet UI
    */
   add(
     outlet: string,
     // tslint:disable-next-line: unified-signatures
-    factory: ComponentFactory<any>,
+    factory: T,
     position?: OutletPosition
   ): void;
   /**
    * @param templateOrFactory A `ComponentFactory` that inserts a component dynamically.
    */
   add(
     outlet: string,
-    templateOrFactory: TemplateRef<any> | ComponentFactory<any>,
+    templateOrFactory: T,
     position: OutletPosition = OutletPosition.REPLACE
   ): void {
     if (position === OutletPosition.BEFORE) {
-      this.templatesRefsBefore.set(outlet, templateOrFactory);
+      this.store(this.templatesRefsBefore, outlet, templateOrFactory);
     }
     if (position === OutletPosition.REPLACE) {
-      this.templatesRefs.set(outlet, templateOrFactory);
+      this.store(this.templatesRefs, outlet, templateOrFactory);
     }
     if (position === OutletPosition.AFTER) {
-      this.templatesRefsAfter.set(outlet, templateOrFactory);
+      this.store(this.templatesRefsAfter, outlet, templateOrFactory);
     }
   }
 
+  /**
+   *
+   * Returns a single object or multiple objects for the given outlet reference,
+   * depending on the `stacked` argument.
+   *
+   * @param outlet The outlet reference
+   * @param position the outlet position, `OutletPosition.before`, `OutletPosition.AFTER` or `OutletPosition.REPLACE`
+   * @param stacked Indicates whether an array of outlet components is returned
+   */
   get(
     outlet: string,
-    position: OutletPosition = OutletPosition.REPLACE
-  ): TemplateRef<any> | ComponentFactory<any> {
+    position: OutletPosition = OutletPosition.REPLACE,
+    stacked = AVOID_STACKED_OUTLETS
```

Comment:
I have a question about the stacked/AVOID_STACKED_OUTLETS. Basically in the one place where it's used we hard code it to true. Is there a use case where someone would want it set to false on purpose?

Also, I'm wondering if maybe this could be a property in the outlet or outletRef. For example at the moment we can specify the position but would it be useful to also say if we want to stack or not?

Reply:
I was thinking about such a feature, but I'm not sure if we should go this way. We currently start to use outlets in our own code base. Multiple modules could use an outlet ref, without necessary knowing, and on top of that, customers could introduce additional outlets for the same outletRef. 

An example is the new `cx-storefront` outlet reference. This is used by ASM, will be used by skipLinks, and could also be used by customers. If the customer would use such a property (say stacked="true"), standard functionality would be missed out.

Reply:
Alright I see, this makes. I hadn't thought about this particular use case.